core/creator/src/main/java/io/quarkus/creator/phase/runnerjar/RunnerJarPhase.java

```diff
@@ -426,6 +431,55 @@ private void filterZipFile(Path resolvedDep, Path targetPath, Set<String> transf
         }
     }
 
+    /**
+     * Handles the core attributes and exception handling for the Manifest.
+     * Setting the Version, Main Class and Classpath
+     *
+     * @param attributes - Manifest Attributes
+     * @param classPath - The Main Class
+     * @throws IOException
+     */
+    private void handleCoreAttributes(final Attributes attributes, final String classPath) {
+        attributes.put(Attributes.Name.MANIFEST_VERSION, "1.0");
+        if (attributes.containsKey(Attributes.Name.CLASS_PATH)) {
+            log.warn(
+                    "Your MANIFEST.MF already defined a CLASS_PATH entry. Quarkus has overwritten this existing entry.");
+        }
+        attributes.put(Attributes.Name.CLASS_PATH, classPath);
+        if (attributes.containsKey(Attributes.Name.MAIN_CLASS)) {
+            String existingMainClass = attributes.getValue(Attributes.Name.MAIN_CLASS);
+            if (!mainClass.equals(existingMainClass)) {
+                log.warn("Your MANIFEST.MF already defined a MAIN_CLASS entry. Quarkus has overwritten your existing entry.");
+            }
+        }
+        attributes.put(Attributes.Name.MAIN_CLASS, mainClass);
+    }
+
+    /**
+     * Resolve any attribute name that is a place holder between ${...} notation.
+     * This is used by build systems that inject metadata to the manifest at build time,
+     * and generally created by CI/CD systems.
+     *
+     * @param attributes - Manifest Attributes to be resolved
+     */
+    private void handleResolvableAttributes(final Attributes attributes) {
+        attributes.keySet().forEach(k -> {
+            String value = (String) attributes.get(k);
+            if (Objects.nonNull(value)) {
+                String trimmedValue = value.trim();
+                if (!trimmedValue.isEmpty() && trimmedValue.startsWith(PLACEHOLDER_PREFIX) &&
```

Comment:
If we are going to do this it should be based on our existing expansion code, and take the value from MP config rather than a system property.

I think something like `org.wildfly.common.expression.Expression.compile(value, Expression.Flag.LENIENT_SYNTAX, Expression.Flag.NO_TRIM).evaluate(ConfigExpander.INSTANCE)`